docs/dev-guides/personal-environments.md

```diff
@@ -469,6 +469,128 @@ gke-terra-integration-default-v2-95d71f95-gdfy   Ready    <none>   30h   v1.17.1
+
+The required namespace permissions are specified in the 3 manifest templates which comes with the Test Runner Library distribution.
+The `setup-k8s-testrunner.sh` script templates the formation of the actual manifests for deploying to a namespace.
+The `setup-k8s-testrunner.sh` script also carries out the following functions:
+
+* Provision the Kubernetes Service Account, RBAC Role and RoleBinding for Test Runner.
+* Export credentials of the Test Runner Kubernetes Service Account to Vault.
+
+To set up a namespace for Test Runner resiliency tests, simply run the command in the following example (`terra-zloery` namespace for example).
+
+The first argument is the `kubectl context` mentioned elsewhere in this document.
+
+The second argument is the Terra namespace (without the `terra-` prefix).
+
+The third argument is just some text to describe the application itself.
+```shell script
+$ ./setup-k8s-testrunner.sh gke_terra-kernel-k8s_us-central1-a_terra-integration zloery workspacemanager
+```
+
+Once the script above ran successfully, the namespace is ready for resiliency testing through the Test Runner Framework.
+
+The Kubernetes credentials stored in Vault needs to be rendered by means of the `./render-k8s-config.sh zloery` script in the test repository before kicking off the resiliency tests.
+```shell script
+$ ./render-k8s-config.sh zloery
+```
+
+<summary>testrunner-k8s-serviceaccount.yml.template</summary>
+
+```text
+# Do not modify this template file.
+
+# This template file is used for setting a Test Runner K8S Service Account
+# for running resiliency tests in a namespace.
+#
+# This template file is to be used in conjunction with the other template files
+#
+#   testrunner-k8s-role.yml.template
+#   testrunner-k8s-rolebinding.yml.template
+#
+# within an automation pipeline and is not meant to be run separately or manually.
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app.kubernetes.io/component: APP
+  name: testrunner-k8s-sa
+  namespace: terra-NAMESPACE
+```
+
+<summary>testrunner-k8s-role.yml.template</summary>
+
+```text
+# Do not modify this template file.
+
+# This template file is used for setting a Test Runner privileged RBAC role
+# for running resiliency tests in a namespace.
+#
+# This template file is to be used in conjunction with the other template files
+#
+#   testrunner-k8s-sa.yml.template
+#   testrunner-k8s-rolebinding.yml.template
+#
+# within an automation pipeline and is not meant to be run separately or manually.
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: testrunner-k8s-role
+  # A k8s namespace: e.g. terra-wsmtest, terra-ichang.
+  # Avoid using default or system namespaces such as kube-system.
+  namespace: terra-NAMESPACE
+  labels:
+    app.kubernetes.io/component: APP
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "pods/exec"]
+    verbs: ["get", "list", "watch", "delete", "patch", "create", "update"]
```

Comment:
do you need all these verbs?

Reply:
Only "create" wasn't used, the others are needed for resiliency tests.